### ISSUE TITLE
Support for hybrid torrent files (bep 0052)

### ISSUE THREAD

**Issue by Lanzaa** [transmission/transmission#3386]
In version 3.00 (bb6b5a062) opening a hybrid v1+v2 torrent file in transmission-qt results in the error: "invalid or corrupt torrent file".

Hybrid torrent files are intended to be backward compatible.

Blocks: #458

**Comment by Lanzaa** [comment]
I think this is worthwhile to get into the next release (4.0?). The current main branch also fails to handle hybrid files.

**Comment by Lanzaa** [comment]
@ckerr I should have a PR for this soon. As a side conversation, how do I follow the 4.0 release process?

**Comment by ckerr** [comment]
> In version 3.00 (bb6b5a0) opening a hybrid v1+v2 torrent file in transmission-qt results in the error: "invalid or corrupt torrent file".

This may be failing in 3.00 but AFAIK hybrid torrents are working correctly in `main` -- this is what the `transmission-show-bittorrent-v2-hybrid-test` test is for? So as far as 3.00 goes I think this issue is a duplicate of #1964?

As far as padding & extended file attributes go, that would be fantastic.

> I should have a PR for this soon. As a side conversation, how do I follow the 4.0 release process?

- 4.0.0-beta.1 will be the first release we've had in years -- the project was (mostly) dormant for a few years until development spun up again in late 2021. Which is to say, there's not a strict process right now.

- The ordinary process of making a PR, addressing review feedback, etc. just as you did in 3380 will apply. There's not a hard deadline for 4.0.0-beta.1 but at a minimum you've got a couple of weeks to get code in.

- Filing issues like this one for "this is what I'm thinking of working on next; WDYT?" is great.

- One thing I'd really like for new code like this is to add test. The testing situation has been slowly getting better over the last 8 months or so, and the "parsing hybrid torrents" topic is in a sweet spot because the existing test structure for `transmission-show` will work as a smoke test for adding support for padding & attribs. If you you want more fine-grained testing -- eg if you're adding attributes to `tr_torrent_metainfo` and want to test that class' contents when fed different torrent files -- it would probably make sense to add tests new into tests/libtransmission/ .

**Comment by Lanzaa** [comment]
Hybrid torrents were failing to download or verify because the padding files were missing. The padding files are not contained in the 'file tree' metadata, but are in 'files'.
